packages/edit-site/src/components/navigation-sidebar/navigation-panel/style.scss

```
@@ -106,3 +112,21 @@
 	padding-top: $grid-unit-05;
 	color: $gray-700;
 }
+
+.edit-site-navigation-panel__template-item-title {
+	display: inline-block;
+	padding-right: $grid-unit-20;
+	position: relative;
+}
+
+.edit-site-navigation-panel__template-item-customized-dot {
+	position: absolute;
+	right: $grid-unit-05;
+	top: 50%;
+	margin-top: -$grid-unit-05;
+	width: $grid-unit-10;
+	height: $grid-unit-10;
+	display: block;
+	background: var(--wp-admin-theme-color);
+	border-radius: 50%;
+}
```

How about inline block?

```suggestion
.edit-site-navigation-panel__template-item-customized-dot {
	background: var(--wp-admin-theme-color);
	border-radius: 50%;
	display: inline-block;
	height: $grid-unit-10;
	margin-left: 4px;
	width: $grid-unit-10;
}
```